src/main.cpp

```diff
@@ -1357,10 +1357,12 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
             dFreeCount += nSize;
         }
 
-        if (fRejectAbsurdFee && nFees > ::minRelayTxFee.GetFee(nSize) * 10000)
-            return error("AcceptToMemoryPool: absurdly high fees %s, %d > %d",
-                         hash.ToString(),
-                         nFees, ::minRelayTxFee.GetFee(nSize) * 10000);
+        if (fRejectAbsurdFee && nFees > ::minRelayTxFee.GetFee(nSize) * 10000) {
+            string errmsg = strprintf("absurdly high fees %s, %d > %d",
+                             hash.ToString(),
+                             nFees, ::minRelayTxFee.GetFee(nSize) * 10000);
+            return state.DoS(0, error(("AcceptToMemoryPool: " + errmsg).c_str()), REJECT_ABSURDLYHIGHFEE, errmsg);
```

Comment:
Hmm.  REJECT_ABSURDLYHIGHFEE is similar to REJECT_INSUFFICIENTFEE which uses state.DoS():
```
                return state.DoS(0, error("AcceptToMemoryPool: not enough fees %s, %d < %d",
                                        hash.ToString(), nFees, txMinFee),
                                REJECT_INSUFFICIENTFEE, "insufficient fee");
```
In both cases, the transactions themselves are valid and could be mined into a block.  If local rules (a form of "emergent consensus") allow for insufficient fee, shouldn't we treat an absurdly high fee the same way?

Reply:
Insufficient fee is a node / standard rule, yes - but one that is communicated over the network precisely because it affects remote users (particularly if the local node has enabled DoS-prevention mechanisms, some of which I think we have, and some of which I know are added in upstream's 0.12).

By contrast, "absurdly high fee" has only ever been a local user protection. There is no rational reason for miners to reject high-fee transactions, and if we do in fact desire a decentralised mining state, no rational reason for any node to reject high-fee transactions. If we want to change this (and explicitly start rejecting high-fee transactions over the network, which we do not currently do), we should do so in its own issue, not as a side-effect of this PR (which is about exposing the error message to the local user).

Reply:
Changed to state.Error to unblock this PR.